cmd/dep/testdata/harness_tests/init/gvt/case1/final/Gopkg.toml

```diff
@@ -5,6 +5,7 @@
 
 [[constraint]]
   name = "github.com/sdboyer/deptestdos"
+  version = "2.0.0"
```

Comment:
tldr: This is expected and isn't a bad side-effect of your change.

There is a typo in the gvt testdata below that caused the import analysis to not find the associated tag for the imported revision (the XXX on the revision cause the rev to not match). When we stopped providing an empty constraint, the solver filled in the gaps.

```
{
      "importpath": "github.com/sdboyer/deptestdos",
      "repository": "https://github.com/sdboyer/deptestdos",
      "revision": "5c607206be5decd28e6263ffffdcee067266015eXXX",
      "branch": "master"
    },
```

The solver for the glide import test did not "fill in the gaps" and generate a constraint because the testdata in that case was using an untagged revision.